parser_test.go

```diff
@@ -1212,3 +1212,59 @@ func TestPointerToList(t *testing.T) {
 // 	require.NoError(t, err)
 // 	require.NotNil(t, ast.List)
 // }
+
+func TestNegation(t *testing.T) {
+	type grammar struct {
+		EverythingUntilSemicolon *[]string `@!';'* @';'`
+	}
+	p := mustTestParser(t, &grammar{})
+	ast := &grammar{}
+	err := p.ParseString(`hello world ;`, ast)
+	require.NoError(t, err)
+	require.Equal(t, &[]string{"hello", "world", ";"}, ast.EverythingUntilSemicolon)
+
+	err = p.ParseString(`hello world`, ast)
+	require.Error(t, err)
+}
+
+func TestNegationWithPattern(t *testing.T) {
+	type grammar struct {
+		EverythingMoreComplex *[]string `@!(';' String)* @';' @String`
+	}
+
+	// Note: we need more lookahead since (';' String) needs some before failing to match
+	//         this is only if we use Stop() which may Accept if we're past the lookahead count...
```

Comment:
`UseLookahead(2)`?

Reply:
No need for that anymore actually since I ignore Stop !